code/parse/sexp.cpp

```diff
@@ -16804,6 +16805,17 @@ int sexp_query_orders (int n)
 	return hud_query_order_issued(order_to, order, target, timestamp, order_from, special);
 }
 
+int sexp_time_to_goal(int n)
+{
+	auto ship_entry = eval_ship(n);
+	if (!ship_entry || !ship_entry->shipp) {
```

Comment:
It's a little tricky to figure out what this particular sexp should return.  I think this makes the most sense:
- if `ship_entry` is null, return `SEXP_NAN`
- if the ship status is NOT_YET_ARRIVED, return `SEXP_CANT_EVAL`
- if it's EXITED, return `SEXP_NAN_FOREVER`.

As for the special cases:
- if the distance is < 1, the sexp should return 0.  It assumes the goal fulfillment is imminent.
- if there is no valid goal, the sexp should return `SEXP_CANT_EVAL`.  This will put directives on pause until the goal is valid and active.

Reply:
As raised on discord, there was a case where `ship_return_seconds_to_goal()` could return -1 to mean "is not moving or can not move" which should also evaluated to `SEXP_CANT_EVAL`